### ISSUE TITLE
Unable to use Let's encrypt with Kura 5

### ISSUE THREAD

**Issue by ctron** [eclipse/kura#3467]
**Describe the bug**

I am struggling again to set up a connection for Kura with an MQTT endpoint which is secured with a certificate signed by Let's encrypt.

This is the same issue as #2594, just for Kura 5 and with the difference that the proposed fix no longer works.

**To Reproduce**
1. Set up a new connect to an MQTT broker using a Let's encrypt signed certificate
2. Add the Let's encrypt root certificate under `Security > Certificates` to the `SSLKeystore`
3. Try to connect

**Expected behavior**

Connects properly.

**Logs**

~~~
2021-07-05T10:44:10,949 [DataServiceImpl:ReconnectTask:org.eclipse.kura.data.DataService] WARN  o.e.k.c.d.DataServiceImpl - Connect failed
org.eclipse.kura.KuraConnectException: "Connection failed. Cannot connect"
        at org.eclipse.kura.core.data.transport.mqtt.MqttDataTransport.connect(MqttDataTransport.java:359)
        at org.eclipse.kura.core.data.DataServiceImpl$2.run(DataServiceImpl.java:606)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: MqttException (0) - javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at org.eclipse.paho.client.mqttv3.internal.ExceptionHelper.createMqttException(ExceptionHelper.java:38)
        at org.eclipse.paho.client.mqttv3.internal.ClientComms$ConnectBG.run(ClientComms.java:736)
        ... 1 more
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.ssl.Alert.createSSLException(Alert.java:131)
        at sun.security.ssl.TransportContext.fatal(TransportContext.java:324)
        at sun.security.ssl.TransportContext.fatal(TransportContext.java:267)
        at sun.security.ssl.TransportContext.fatal(TransportContext.java:262)
        at sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:654)
        at sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:473)
        at sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:369)
        at sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:377)
        at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:444)
        at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:422)
        at sun.security.ssl.TransportContext.dispatch(TransportContext.java:182)
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:152)
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1383)
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1291)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:435)
        at org.eclipse.paho.client.mqttv3.internal.SSLNetworkModule.start(SSLNetworkModule.java:147)
        at org.eclipse.paho.client.mqttv3.internal.ClientComms$ConnectBG.run(ClientComms.java:722)
        ... 1 more
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:456)
        at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:323)
        at sun.security.validator.Validator.validate(Validator.java:271)
        at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:315)
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:223)
        at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:129)
        at sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:638)
        ... 13 more
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
        at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
        at java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280)
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:451)
        ... 19 more
~~~

**Target Environment (please complete the following information):**
 - Container
 - OS version: UBI8 image, `5.12.12-300.fc34.x86_64 #1 SMP Fri Jun 18 14:30:51 UTC 2021`
 - Additional info: Java: `OpenJDK Runtime Environment 1.8.0_292-b10`

**Additional context**

**Comment by marcellorinaldo** [comment]
Hi @ctron!
The correct steps, assuming you want to connect to `mqtt.eclipseprojects.io:8883` (note that `mqtts://iot.eclipse.org:8883/ `is no longer available), are:
1. login as `admin/admin`
2. from a terminal, retrieve the certificate with: `openssl s_client -showcerts -connect mqtt.eclipseprojects.io:8883`
3. from Security section, remove the default `ssl-eclipse` certificate (it expired in 2019)
4. create the new certificate (SSLKeystore) with alias `lets-encrypt`
5. got to the cloud connections and edit the MqttDataTransport section by setting `Broker.url = mqtts://mqtt.eclipseprojects.io:8883`, leave everything else as it is
6. apply and then connect

**Comment by ctron** [comment]
>     4\. create the new certificate (SSLKeystore) with alias `lets-encrypt`

And here is the problem, also see #3468.

There are two Let's Encrypt (LE) root certificates. One cross signed, the other not. You need to know which one to pick. Additionally, the cross-signed one has a trailing whitespace in the header line, which makes the parser of Kura fail, without giving any reasonable error information.

So what you have to do is:

* Download the cross signed cert from here https://letsencrypt.org/certs/isrg-root-x1-cross-signed.pem
* Remove the trailing space on the first line
* Add it to Kura

I think that process should be improved in two ways:

1. Fix the issues described in #3468 
2. Provide an "out of the box" SSLKeyStore with the LE certificates. Either as an additional one, or as the default one. I think there is no value in providing the old Eclipse MQTT certificate and not having LE on board.

**Comment by marcellorinaldo** [comment]
The most recent Eclipse certificate has been added with PR #3465. More robust checks are implemented in PR #3469.
